src/app/components/Feed/Feed.tsx

```diff
@@ -1,32 +1,46 @@
-import { useState } from 'react';
+import { MatchClient, QueryOptions } from '../../../types';
+import { useEffect, useState } from 'react';
 import useFetch from '../../hooks/useFetch';
 import { getMatches } from '../../utils/api';
 import MatchItem from '../MatchItem/MatchItem';
 import classes from './Feed.module.css';
 
 type FeedProps = {
+  username: MatchClient['username'] | null;
   onMatchClick: (matchId: string) => void;
 };
-function Feed({ onMatchClick }: FeedProps): JSX.Element {
-  const [page, setPage] = useState(1);
+function Feed({ username, onMatchClick }: FeedProps): JSX.Element {
+  const [query, setQuery] = useState<QueryOptions>({
+    page: 1,
+    itemsPerPage: 10,
+  });
+  const { page } = query;
 
-  const { data: matches } = useFetch(
-    () => getMatches({ page, itemsPerPage: 10 }),
-    {
-      invalidateOn: [page],
-      refreshInterval: 60000,
+  useEffect(() => {
+    if (username) setQuery({ ...query, page: 1, username });
```

Comment:
This effect doesn't depend on `query`, which might lead to unexpected results.
Actually, in this case it shouldn't fail, but to make sure it won't fail in the future:

```suggestion
    if (username) setQuery(query => ({ ...query, page: 1, username }));
```

In this case, you are always sure that you have the most recent `query`.
Same issue in L:24.
https://reactjs.org/docs/hooks-reference.html#usestate